jsurfer-core/src/main/java/org/jsfr/json/filter/Type.java

```diff
@@ -0,0 +1,71 @@
+package org.jsfr.json.filter;
+
+import org.jsfr.json.provider.JsonProvider;
+
+/**
+ * Types that `type()` method can return. See {@link ItemMethod#TYPE}.
+ */
+public enum Type {
+    /**
+     *
+     */
+    NULL {
+        @Override
+        public boolean isInstanceOf(Object obj, JsonProvider<?, ?, ?> provider) {
+            return provider.isPrimitiveNull(obj);
+        }
+    },
+    /**
+     *
+     */
+    NUMBER {
+        @Override
+        public boolean isInstanceOf(Object obj, JsonProvider<?, ?, ?> provider) {
+            return provider.isPrimitiveNumber(obj);
+        }
+    },
+    /**
+     *
+     */
+    STRING {
+        @Override
+        public boolean isInstanceOf(Object obj, JsonProvider<?, ?, ?> provider) {
+            return provider.isPrimitiveString(obj);
+        }
+    },
+    /**
+     *
+     */
+    BOOLEAN {
+        @Override
+        public boolean isInstanceOf(Object obj, JsonProvider<?, ?, ?> provider) {
+            return provider.isPrimitiveBoolean(obj);
+        }
+    },
+    /**
+     *
+     */
+    ARRAY,
+    /**
+     *
+     */
+    OBJECT;
+
+    public boolean isInstanceOf(Object obj, JsonProvider<?, ?, ?> provider) {
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return name().toLowerCase();
+    }
+
+    public static Type from(String value) {
+        for (Type type : values()) {
+            if (type.toString().equalsIgnoreCase(value)) {
```

Comment:
The matching shouldn't be case-insensitive I think. The `type.toString()` already converts to lower case, so we can do just `equals` here.
But perhaps we can create a static `HashMap<String, Type>` that we can query here, to avoid converting all enum names to lower case each time. Or a `switch` could work.

Reply:
Done